src/material/slider/slider-input.ts

```diff
@@ -0,0 +1,714 @@
+   *
+   * This flag is used to avoid flickering
+   * when correcting values on pointer up/down.
+   */
+  _skipUIUpdate: boolean = false;
+
+  /** Callback called when the slider input value changes. */
+  private _onChangeFn: (value: any) => void = () => {};
+
+  /** Callback called when the slider input has been touched. */
+  private _onTouchedFn: () => void = () => {};
+
+  constructor(
+    readonly _ngZone: NgZone,
+    readonly _elementRef: ElementRef<HTMLInputElement>,
+    readonly _cdr: ChangeDetectorRef,
+    @Inject(MAT_SLIDER_TOKEN) protected _slider: MatSliderInterface,
+  ) {
+    this._hostElement = _elementRef.nativeElement;
+    this._ngZone.runOutsideAngular(() => {
+      this._hostElement.addEventListener('pointerdown', this._onPointerDown);
+      this._hostElement.addEventListener('pointermove', this._onPointerMove);
+      this._hostElement.addEventListener('pointerup', this._onPointerUp);
+    });
+  }
+
+  ngOnDestroy(): void {
+    this._hostElement.removeEventListener('pointerdown', this._onPointerDown);
+    this._hostElement.removeEventListener('pointermove', this._onPointerMove);
+    this._hostElement.removeEventListener('pointerup', this._onPointerUp);
+    this._destroyed.next();
+    this._destroyed.complete();
+  }
+
+  initProps(): void {
+    this._updateWidthInactive();
+    this.disabled = this._slider.disabled;
+    this.step = this._slider.step;
+    this.min = this._slider.min;
+    this.max = this._slider.max;
+    this._initValue();
+  }
+
+  initUI(): void {
+    this._updateThumbUIByValue();
+  }
+
+  _initValue(): void {
+    this._hasSetInitialValue = true;
+    if (this._initialValue === undefined) {
+      this.value = this._getDefaultValue();
+    } else {
+      this.value = this._initialValue;
+    }
+  }
+
+  _getDefaultValue(): number {
+    return this.min;
+  }
+
+  _onBlur(): void {
+    this._setIsFocused(false);
+    this._onTouchedFn();
+  }
+
+  _onFocus(): void {
+    this._setIsFocused(true);
+  }
+
+  _onChange(): void {
+    // only used to handle the edge case where user
+    // mousedown on the slider then uses arrow keys.
+    if (this._isActive) {
+      this._updateThumbUIByValue({withAnimation: true});
+    }
+  }
+
+  _onInput(): void {
+    this.valueChange.emit(this._hostElement.value);
+    this._onChangeFn(this.value);
+    // handles arrowing and updating the value when
+    // a step is defined.
+    if (this._slider.step || !this._isActive) {
+      this._updateThumbUIByValue({withAnimation: true});
+    }
+    this._slider._onValueChange(this);
+  }
+
+  _onNgControlValueChange(): void {
+    // only used to handle when the value change
+    // originates outside of the slider.
+    if (!this._isActive || !this._isFocused) {
+      this._slider._onValueChange(this);
+      this._updateThumbUIByValue();
+    }
+    this._slider.disabled = this._formControl!.disabled;
+  }
+
+  _onPointerDown = (event: PointerEvent): void => {
```

Comment:
You can convert the `_pointerDownHandler` into an arrow function which will allow you to delete this one. The same goes for the other event listeners.

Reply:
Using an arrow function prevents it from getting overridden properly. My latest change is another workaround I came up with for this